zerver/lib/actions.py

```diff
@@ -1731,6 +1732,10 @@ def is_service_bot(row: Dict[str, Any]) -> bool:
 
     service_bot_tuples = [(row["id"], row["bot_type"]) for row in rows if is_service_bot(row)]
 
+    # We also need the user IDs of all bots, to avoid trying to send push/email
```

Comment:
Do we want to update the comment on line 1719 which talks about "These **two** bot data structures..." now that there are three of them?

Reply:
Yes, thanks for pointing out. Done.